cmd/crictl/version.go

```diff
@@ -48,10 +45,10 @@ var runtimeVersionCommand = &cli.Command{
 }
 
 // Version sends a VersionRequest to the server, and parses the returned VersionResponse.
-func Version(client pb.RuntimeServiceClient, version string) error {
+func Version(client internalapi.RuntimeService, version string) error {
 	request := &pb.VersionRequest{Version: version}
```

Comment:
I wonder what is the history of this field. Both Containerd and Cri-O seems to be simply ignoring this value

Reply:
I think it was for the Docker API version, no ?

```go
        typedVersion, err := r.runtimeClient.Version(ctx, &runtimeapi.VersionRequest{
                Version: apiVersion,
        })
```

```
docker@minikube:~$ sudo crictl version
Version:  0.1.0
RuntimeName:  docker
RuntimeVersion:  20.10.12
RuntimeApiVersion:  1.41.0
```

Reply:
@SergeyKanzhelev : apparently the apiVersion was used to make the Docker API Version semver-compatible:

```go
       // Docker API version (e.g., 1.23) is not semver compatible. Add a ".0"
       // suffix to remedy this.
       apiVersion := fmt.Sprintf("%s.0", v.APIVersion)
```

https://github.com/kubernetes/kubernetes/commit/13b23c51cb8539ec83e17474618b5ecb34f5bf28

Then it was moved out to a separate track, but of course the field remained in the API (not used for Docker either).

https://github.com/kubernetes/kubernetes/commit/d8e29e782f4087f810ae543d038772c503ce2d4b